presto-main/src/main/java/com/facebook/presto/operator/SimpleJoinProbe.java

```diff
@@ -140,4 +140,14 @@ public Page getPage()
     {
         return page;
     }
+
+    @Override
+    public Page buildDictionaryPage(int[] indices, PageBuilder sourcePageBuilder)
+    {
+        int[] outputChannels = new int[probeOutputChannels.size()];
+        for (int i = 0; i < outputChannels.length; i++) {
+            outputChannels[i] = probeOutputChannels.get(i);
```

Comment:
You can just copy list into array with:
```
int[] outputChannels = probeOutputChannels.stream()
                .mapToInt(i -> i)
                .toArray();
```
(if you really can't just pass `probeOutputChannels` by reference, ensuring it's immutability in ctor first)